packages/destination-actions/src/destinations/cordial/addProductToCart/__tests__/index.test.ts

```
@@ -0,0 +1,46 @@
+import nock from 'nock'
+import { createTestEvent, createTestIntegration } from '@segment/actions-core'
+import Destination from '../../index'
+
+const testDestination = createTestIntegration(Destination)
+
+describe('Cordial.addProductToCart', () => {
+  it('should work with default mappings', async () => {
+    nock(/api.cordial.io/).post('/api/segment/addProductToCart').reply(200, {})
+
+    const event = createTestEvent({
+      event: 'Product Added',
+      userId: 'abc123',
+      timestamp: '1631210000',
+      properties: {
+        product_id: '51easf12',
+        sku: 'TEST-SKU',
+        quantity: 2,
+        category: '51easf12',
+        name: 'TEST-SKU',
+        price: 2.35,
+        url: 'https://example.com/product1',
+        image_url: 'https://cdn.example.com/product.jpg',
+        brand: 'Test brand',
+        variant: 'black',
+        coupon: 'APPLY50%',
+      }
+    })
+
+    const mapping = {
+      userIdentities: {'channels.email.address': 'contact@example.com'}
+    }
+
+    const settings = {
+      apiKey: 'cordialApiKey',
+      endpoint: 'https://api.cordial.io' as const
+    }
+
+    await testDestination.testAction('addProductToCart', {
```

It would be good to have at least one line of code validating behaviour in the testcase to protect against silent failures. 

This can be done by:
1.  Receiving the `response` object from this call and adding a few `expect` statements on it.
2. Validating that the endpoint got called (Nock [example](https://github.com/segmentio/action-destinations/blob/0ad65009907201bb717ca43e762aff457f77748a/packages/destination-actions/src/destinations/salesforce/__tests__/sf-operations.test.ts#L14); Mock [example](https://github.com/segmentio/action-destinations/blob/4c3eb45f5d92775cddb37687184a097904633705/packages/destination-actions/src/destinations/google-sheets/__tests__/postSheet.operations.test.ts#L81))

---

Same comment applies to removeProductFromCart and upsertOrder.

---

@rhall-twilio I've pushed an update to the test suite, please check if I've got you right.

Also, I've noticed that in test suites (e.g. packages/destination-actions/src/destinations/cordial/addProductToCart/__tests__/index.test.ts:121) not mapped properties are returned as undefined. Optional properties are nullable in generated-types, and while checking Dev Action Center I see that those properties are filtered out from request. On the other hand, when I put mapping in Dev Action Tester and then remove it, empty string is added as property.

Could you confirm that we would observe the same behavior  as in Dev Action Tester on Segment's production? Meaning that all null properties will be automatically filtered out from request and won't being sent, while empty objects or arrays will be preserved? Also, is there a way for a customer to remove mapping without property being set as empty string (so optional property would be null)?

---

@rhall-twilio ping, just to check if above comment was seen

---

@dlasky for better context

---

@rhall-twilio Could you please provide an ETA for when this PR may be merged?

---

there were open issues so I wasn't sure if we were 100 to merge when I was looking at it.

---

@dlasky Are there any right now? Please suggest what I should do to get this merged, thanks in advance!